lib/ddtrace/contrib/rake/instrumentation.rb

```diff
@@ -15,19 +15,21 @@ module InstanceMethods
           def invoke(*args)
             return super unless enabled?
 
-            tracer.trace(SPAN_NAME_INVOKE) do |span|
+            tracer.trace(SPAN_NAME_INVOKE, span_options) do |span|
               super
               annotate_invoke!(span, args)
             end
+            tracer.shutdown! if tracer.active_span.nil? && ::Rake.application.top_level_tasks.include?(name)
           end
 
           def execute(args = nil)
             return super unless enabled?
 
-            tracer.trace(SPAN_NAME_EXECUTE) do |span|
+            tracer.trace(SPAN_NAME_EXECUTE, span_options) do |span|
               super
               annotate_execute!(span, args)
             end
+            tracer.shutdown! if tracer.active_span.nil? && ::Rake.application.top_level_tasks.include?(name)
```

Comment:
Looks good! 
It would be great to have some tests for this new behavior. 

I think also this could be put in `ensure` block so that even when task throws an exception we will still be able to flush the trace with info about that exception.